x/tokenregistry/keeper/keeper.go

```diff
@@ -85,75 +85,67 @@ func (k keeper) CheckDenomPermissions(ctx sdk.Context, denom string, requiredPer
 	return true
 }
 
+func (k keeper) GetDenomPrefix(ctx sdk.Context, denom string) []byte {
+	return append(types.TokenDenomPrefix, []byte(denom)...)
+}
+
 func (k keeper) GetDenom(ctx sdk.Context, denom string) types.RegistryEntry {
-	wl := k.GetRegistry(ctx)
-
-	// TODO: This is inefficient, and potentially an attack vector. Investigate
-	// using a map instead
-	// See ticket #2065
-	for i := range wl.Entries {
-		if wl.Entries[i] != nil && strings.EqualFold(wl.Entries[i].Denom, denom) {
-			return *wl.Entries[i]
-		}
-	}
 
-	return types.RegistryEntry{
-		IsWhitelisted: false,
-		Denom:         denom,
-	}
+	var entry types.RegistryEntry
```

Comment:
GetDenom needs to return a valid struct if there's no key stored for the denom

Reply:
accepted.